src/app/settings/system-config/global-config/global-config.component.ts

```diff
@@ -0,0 +1,127 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, FormControl, Validators } from '@angular/forms';
+import { MAT_DATE_FORMATS } from '@angular/material/core';
+import { passwordMatchingValidatior } from 'src/app/directives/confirm-passowrd.directive';
+import { SettingsService } from 'src/app/services/settings.service';
+import { DateUtil } from 'src/app/services/utility/DateUtil';
+import { ISettingItem } from '../settings-model';
+
+@Component({
+  selector: 'app-global-config',
+  templateUrl: './global-config.component.html',
+  styleUrls: ['./global-config.component.scss'],
+
+})
+export class GlobalConfigComponent implements OnInit {
+
+
+  @Input('globalSettings') globalSettings: ISettingItem[];
+
+
+  dateFormatArray: string[] = [DateUtil.DATE_FORMAT, DateUtil.DATE_FORMAT_SHORT];
+  timeFormatArray: string[] = [DateUtil.TIME_FORMAT, DateUtil.TIME_FORMAT_24_HOURS];
+  timestampFormatArray: string[] = [DateUtil.TIME_FORMAT_24_HOURS];
+  selectedDateFormat: string;
+  selectedTimeFormat: string;
+  selectedTimestampFormat: string;
+
+  globalForm = new FormGroup({
+    DATE_FORMAT: new FormControl(''),
+    TIME_FORMAT: new FormControl(''),
+    TIMESTAMP_FORMAT: new FormControl(''),
+    NO_OF_ACTIVE_USERS: new FormControl(10),
+    VALIDITY_TILL: new FormControl(''),
+    VALIDITY_EXPIRY_MSG: new FormControl(''),
+
+  });
+
+  constructor(private settingService: SettingsService, private formBuilder: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.fillForm();
```

Comment:
we should put this in ngOnChanges and only when we have globalSettings value, we can call fillForm. It might happen that we are yet to have any value in globalSettings on ngOnInit.

Reply:
Agreed, but I am only rendering components after we get all the settings from backend. Check system-config ngOnInit